core/pl/stream.pm.sdoc

```diff
@@ -59,6 +59,7 @@ sub rw(&) {my @r = ($_); push @r, $_ while  defined rl && &{$_[0]}; push @q, $_
 sub ru(&) {my @r = ($_); push @r, $_ until !defined rl || &{$_[0]}; push @q, $_ if defined $_; @r}
 sub re(&) {my ($f, $i) = ($_[0], &{$_[0]}); rw {&$f eq $i}}
 BEGIN {ceval sprintf 'sub re%s() {re {%s}}', $_, $_ for 'a'..'l'}
+sub rn($) {my $max_val = shift; my @r = ($_); push @r, $_ while defined rl && (@r < $max_val); push @q, $_ if defined $_; @r} 
```

Comment:
ah, look at that, I never noticed the other `rl` syntax. I'll spike this. Looks like it also `b_ rl 100` might have a bug though: `ni i[a 1] i[a 2] i[b 3] i[a 4] p'r F_ for(0..1000)' p'$a = a; @b = b_ rl 100; r $a, uniq @b, $#b'` only puts 99 elements into `@b`.